app/controllers/RiderController.java

```diff
@@ -19,14 +27,101 @@ public RiderController(RiderRepository riderRepository) {
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
     }
 
-    public CompletionStage<Result> getRiders() {
-        /*return riderRepository.list().thenApplyAsync(riderStream -> {
-           return ok(toJson(riderStream.collect(Collectors.toList())));
-        });*/
-        return null;
+    public CompletionStage<Result> getRider(int riderId){
+        return riderRepository.getRider(riderId).thenApplyAsync(rider -> {
+            return ok(rider);
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "NoResultException":
+                    res = badRequest("No rider with id: " + riderId + " is available in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> addRider(){
+        JsonNode json = request().body().asJson();
+        return parseRider(json).thenApply(rider -> riderRepository.addRider(rider)).thenApply(rider -> {
+            return ok(rider + " has been added");
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "NullPointerException":
+                    res = badRequest("json format of rider was wrong");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    private CompletableFuture<Rider> parseRider(JsonNode json){
+        CompletableFuture<Rider> completableFuture
+                = new CompletableFuture<>();
+
+        Executors.newCachedThreadPool().submit(() -> {
+            try{
+                Rider rider = new Rider();
+                rider.setCountry(json.findPath("country").textValue());
+                rider.setUnkown(json.findPath("isUnkown").booleanValue());
+                rider.setName(json.findPath("name").textValue());
+                rider.setRiderId(json.findPath("riderId").intValue());
+                rider.setStartNr(json.findPath("startNr").intValue());
+                rider.setTeamName(json.findPath("teamName").textValue());
+                rider.setTeamShortName(json.findPath("teamShortName").textValue());
+                completableFuture.complete(rider);
+                return rider;
+            } catch (Exception e){
+                completableFuture.obtrudeException(e);
+                throw  e;
+            }
+        });
+
+        return completableFuture;
+    }
+
+
+    public CompletionStage<Result> deleteAllRiders(){
+        return riderRepository.deleteAllRiders().thenApply(riders -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)